File: bot/logs/logger.js
export function errorLog(error, interaction) {
    sendConsoleLogMessage(error, interaction, 'ERROR');
}

export function infoLog(message, interaction) {
    sendConsoleLogMessage(message, interaction, 'INFO');
}

export function debugLog(message, interaction) {
    sendConsoleLogMessage(message, interaction, 'DEBUG');
}

export function warnLog(message, interaction) {
    sendConsoleLogMessage(message, interaction, 'WARN');
}

function sendConsoleLogMessage(message, interaction, level) {
    if (!interaction) return;
    const userId = interaction.user.id;

    const data = {
        level: level,
        message: message,
        userId: `<@${userId}>`,
        timestamp: new Date().toISOString()
    };

    if (level === 'ERROR') {
        console.error(message);
    } else {
        console.log(JSON.stringify(data, null, 2));
    }
}

File: backend/internal/infrastructure/logging/logger.go
package logging

import "log/slog"

type SLogAdapter struct {
	logger *slog.Logger
}

func NewSLogAdapter() *SLogAdapter {
	return &SLogAdapter{logger: slog.Default()}
}

func (l *SLogAdapter) Debug(message string, fields ...interface{}) {
	l.logger.Debug(message, fields...)
}

func (l *SLogAdapter) Error(message string, fields ...interface{}) {
	l.logger.Error(message, fields...)
}

func (l *SLogAdapter) Info(message string, fields ...interface{}) {
	l.logger.Info(message, fields...)
}

func (l *SLogAdapter) Warn(message string, fields ...interface{}) {
	l.logger.Warn(message, fields...)
}
